src/SFML/Graphics/Image.cpp

```diff
@@ -126,8 +122,8 @@ bool Image::loadFromFile(const std::string& filename)
 
     #else
 
-        priv::ResourceStream stream(filename);
-        return loadFromStream(stream);
+        m_stream = std::make_shared<priv::ResourceStream>(filename);
+        return loadFromStream(*std::static_pointer_cast<priv::ResourceStream*>(m_stream));
```

Comment:
I have the feeling this could be simpler …

Reply:
No idea how to make it simpler. It's a bit confusing, but I can deal with it.

Reply:
Actually when looking at original code https://github.com/SFML/SFML/blob/master/src/SFML/Graphics/Image.cpp#L129
it looks wrong so I restored